Assets/Scripts/Models/Furniture.cs

```diff
@@ -366,6 +357,28 @@ protected bool DEFAULT__IsValidPosition(Tile t)
         return true;
     }
 
+
+    public bool HasPowerCheck()
+    {
+
+        if (powerValue > 0 && isPowerGenerator == false)
```

Comment:
I would like to see that method renamed to just `HasPower`, the `Check` suffix makes me think that this method does something that has side effects (setting variables).